core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -387,4 +393,55 @@ public void dataFilesCleanup() throws IOException {
     Assert.assertTrue("FILE_A should be deleted", deletedFiles.contains(FILE_A.path().toString()));
     Assert.assertTrue("FILE_B should be deleted", deletedFiles.contains(FILE_B.path().toString()));
   }
+  @Test
+  public void testWithExpiringWapThenCherrypick() {
+    // first WAP commit
+    table.newAppend()
+            .appendFile(FILE_B)
```

Comment:
Continuing indentation should be 4 spaces from the start of the previous line, not 8.